core/deployment/src/main/java/io/quarkus/deployment/util/ServiceUtil.java

```diff
@@ -83,4 +90,8 @@ private static void readStream(final Set<String> classNames, final BufferedReade
             classNames.add(line);
         }
     }
+
+    private static String getServiceFileName(Class<?> serviceClass) {
```

Comment:
minor nitpick: this API encourages to initialize the serviceClass Class during augmentation.

Probably not a problem in most cases, but I feel like this shouldn't be encouraged.

Reply:
My opinion is that using strings introduces some fragility I would prefer to avoid.

But I can see your point too. I removed this commit altogether.